07.ls_object/ls.rb

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require './ls_file'
+require 'optparse'
+
+class LS
+  MAX_ROW = 3
+
+  def main
+    opt = OptionParser.new
+
+    params = { dot_match: false, long_format: false, reverse: false }
+    opt.on('-a') { |v| params[:dot_match] = v }
+    opt.on('-l') { |v| params[:long_format] = v }
+    opt.on('-r') { |v| params[:reverse] = v }
+    opt.parse!(ARGV)
+    path = '.'
+
+    collected_filenames = collect_files(path, params[:dot_match], params[:reverse])
+    puts params[:long_format] ? list_long(collected_filenames) : list_short(collected_filenames)
+  end
+
+  private
+
+  def collect_files(path, dot_match, reverse)
+    all_filenames = Dir.foreach(path)
+    filenames = dot_match ? all_filenames : all_filenames.reject { |file| file.start_with?('.') }
```

Comment:
```suggestion
    filenames = Dir.foreach(path)
    filtered_filenames = dot_match ? all_filenames : all_filenames.reject { |file| file.start_with?('.') }
```

「ただのfilenames」から「取捨選択されたfilenames」に変わった、みたいな命名の方が良いかもしれません

Reply:
修正しました